src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -110,17 +110,36 @@ class FilterOptionsConfig {
         config.previousOptions = [];
       }
     }
-    let selectedOptions = config.previousOptions || [];
-    this.options = this.setDefaultSelectedValues(this.options, selectedOptions);
+    const previousOptions = config.previousOptions || [];
+    this.options = this.setDefaultSelectedValues(this.options, previousOptions);
   }
 
-  setDefaultSelectedValues (options, selectedOptions) {
-    return options.map(o => ({
-      ...o,
-      selected: selectedOptions.length
-        ? selectedOptions.includes(o.label)
-        : o.selected
-    }));
+  /**
+   * Sets selected options on load based on options stored in persistent storage,
+   * and if persistent storage is empty then the appliedOnLoad flag.
+   * @param {Array<Object>} options
+   * @param {Array<string>} previousOptions
+   * @returns {Array<Object>}
+   */
+  setDefaultSelectedValues (options, previousOptions) {
+    if (previousOptions.length) {
+      return options.map(o => ({
+        ...o,
+        selected: previousOptions.length
+          ? previousOptions.includes(o.label)
+          : o.selected
+      }));
+    }
+    for (let o of options) {
```

Comment:
We should be consistent in this method about how we're iterating through options and returning a result. Above, we were using Array.map, but here we're using a for loop. I'm partial to the .map way of doing things since it doesn't modify the options param itself. I think a .forEach could work here as well. That would let you break early and we could still use the more functional style of iteration. With a forEach though, I think we'd want to be operating on a copy of options.

Reply:
Refactored!